substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageGeneratorRunner.java

```diff
@@ -169,7 +170,17 @@ public static ImageClassLoader installNativeImageClassLoader(String[] classpath,
          */
         NativeImageGenerator.setSystemPropertiesForImageEarly();
 
-        return new ImageClassLoader(NativeImageGenerator.defaultPlatform(nativeImageClassLoader), nativeImageClassLoaderSupport);
+        Platform targetPlatform = NativeImageGenerator.getTargetPlatform(nativeImageClassLoader);
+        Platform hostPlatform = NativeImageGenerator.getHostPlatform();
+
+        if (System.getProperty("svm.targetPlatform") != null) {
+            /*
+             * This check is only meaningful for default platforms that are supported by --target
+             * parameter (see comment in NativeImageGenerator.getPlatformOS for more information)
+             */
+            UserError.guarantee(NativeImageGenerator.isSameOS(targetPlatform, hostPlatform), "Cross OS compilation isn't supported at the moment.");
```

Comment:
What about android? Shouldn't we just check if libs are present and say target not supported?

Reply:
As for now, we are building Android as `LINUX_AARCH64`. My reasoning is that when Android gets implemented as a `Platform` (2 PR's down the line), it should be extended from `LINUX`, which would still satisfy this check.

Checking for static libraries is already done in `NativeLibraries.java:336`